framework/src/mesh/MooseMesh.C

```diff
@@ -435,6 +442,88 @@ MooseMesh::update()
   _face_info_dirty = true;
 }
 
+void
+MooseMesh::buildLowerDMesh()
+{
+  auto & mesh = getMesh();
+
+  // remove existing lower-d element first
+  std::set<Elem *> deleteable_elems;
+  for (auto & elem : mesh.element_ptr_range())
+    if (elem->subdomain_id() == Moose::INTERNAL_SIDE_LOWERD_ID ||
+        elem->subdomain_id() == Moose::BOUNDARY_SIDE_LOWERD_ID)
+      deleteable_elems.insert(elem);
+
+  for (auto & elem : deleteable_elems)
+    mesh.delete_elem(elem);
+
+  deleteable_elems.clear();
+
+  dof_id_type max_elem_id = mesh.max_elem_id();
+  unique_id_type max_unique_id = mesh.parallel_max_unique_id();
+
+  std::vector<Elem *> side_elems;
+  _higher_d_elem_side_to_lower_d_elem.clear();
+  for (const auto & elem : mesh.active_local_element_ptr_range())
+  {
+    for (unsigned short int side = 0; side < elem->n_sides(); side++)
+    {
+      Elem * neig = elem->neighbor_ptr(side);
+
+      bool build_side = false;
+      if (!neig)
+        build_side = true;
+      else
+      {
+        if (!neig->active())
+          build_side = true;
+        else if (neig->level() == elem->level() && elem->id() < neig->id())
+          build_side = true;
+      }
+
+      if (build_side)
+      {
+        std::unique_ptr<Elem> side_elem(elem->build_side_ptr(side, false));
+
+        // The side will be added with the same processor id as the parent.
+        side_elem->processor_id() = elem->processor_id();
+
+        // Add subdomain ID
+        if (neig)
+          side_elem->subdomain_id() = Moose::INTERNAL_SIDE_LOWERD_ID;
+        else
+          side_elem->subdomain_id() = Moose::BOUNDARY_SIDE_LOWERD_ID;
+
+        // Also assign the side's interior parent, so it is always
+        // easy to figure out the Elem we came from.
+        side_elem->set_interior_parent(elem);
+
+        // Add id
+        side_elem->set_id(max_elem_id++);
+        side_elem->set_unique_id(max_unique_id++);
+
+        side_elems.push_back(side_elem.release());
+
+        // add link between higher d element to lower d element
+        auto pair = std::make_pair(elem, side);
+        auto link = std::make_pair(pair, side_elems.back());
+        auto ilink = std::make_pair(side_elems.back(), side);
+        _lower_d_elem_to_higher_d_elem_side.insert(ilink);
+        _higher_d_elem_side_to_lower_d_elem.insert(link);
+      }
+    }
+  }
+
+  // Finally, add the lower-dimensional element to the Mesh.
+  for (auto & elem : side_elems)
+    mesh.add_elem(elem);
+
+  _mesh_subdomains.insert(Moose::INTERNAL_SIDE_LOWERD_ID);
+  mesh.subdomain_name(Moose::INTERNAL_SIDE_LOWERD_ID) = "INTERNAL_SIDE_LOWERD_SUBDOMAIN";
+  _mesh_subdomains.insert(Moose::BOUNDARY_SIDE_LOWERD_ID);
+  mesh.subdomain_name(Moose::BOUNDARY_SIDE_LOWERD_ID) = "BOUNDARY_SIDE_LOWERD_SUBDOMAIN";
+}
```

Comment:
Please call `mesh.set_isnt_prepared()` at the end of this method in case it's of importance to later users of the mesh

Reply:
What is going to happen if we reparation the high dimensional mesh? The lower dimensional mesh  will be rebuilt automatically?

Reply:
Sounds like we should not hit repartitioning scenarios with the errors that have been inserted and with the placement of this capability in the overall mesh generation sequence. My concern over mesh preparedness has been address so from my perspective this thread is resolved